File: template_tool.py
from pathlib import Path
from string import Template
from check_file_permission import check_file_permission

def template_file(file_path, template_name, substitutions, config):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    templates = {"python_script": "def main():\n    print('Hello, $name')"}
    template = Template(templates.get(template_name, ""))
    content = template.substitute(substitutions)
    file_path = Path(file_path).resolve()
    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)
    return f"Wrote template to {file_path}"

File: ts-tools/template_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function templateFile(
  filePath: string,
  templateName: string,
  substitutions: { [key: string]: string },
  config: any,
): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'write', config)
      .then(() => {
        const templates: { [key: string]: string } = {
          python_script: "def main():\n    print('Hello, $name')",
        };
        const template = templates[templateName];
        if (!template) {
          reject(new Error(`Template not found: ${templateName}`));
          return;
        }
        let content = template;
        for (const key in substitutions) {
          content = content.replace(`$${key}`, substitutions[key]);
        }
        const resolvedPath = path.resolve(filePath);
        fs.writeFile(resolvedPath, content, 'utf-8', (err) => {
          if (err) {
            reject(err);
          } else {
            resolve(`Wrote template to ${filePath}`);
          }
        });
      })
      .catch(reject);
  });
}
